libraries/Matter/src/MatterEndpoints/MatterHumiditySensor.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 Espressif Systems (Shanghai) PTE LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <sdkconfig.h>
+#ifdef CONFIG_ESP_MATTER_ENABLE_DATA_MODEL
+
+#include <Matter.h>
+#include <MatterEndPoint.h>
+
+class MatterHumiditySensor : public MatterEndPoint {
+public:
+  MatterHumiditySensor();
+  ~MatterHumiditySensor();
+  // begin Matter Humidity Sensor endpoint with initial float humidity percent
+  bool begin(double humidityPercent = 0.00) {
+  if (humidityPercent < 0.0 || humidityPercent > 100.0) {
+    log_e("Humidity Sensor Percentage value out of range [0..100].");
+    return false;
+  }
+    return begin(static_cast<uint16_t>(humidityPercent * 100.0f));
+  }
+  // this will just stop processing Humidity Sensor Matter events
+  void end();
+
+  // set the humidity percent with 1/100th of a percent precision
+  bool setHumidityPercent(double humidityPercent) {
```

Comment:
similar to the temperature, humidity is "always" in percents, so leaving just `setHumidity` and `getHumidity` should be enough (and consistent with the temperature sensor).

Reply:
done